### PR TITLE
Cms

### PR DESCRIPTION
**Pull request by ShiroUsagi-san**
Updating cms routes

### CODE REVIEW

File: insalan/cms/tests.py [hunk 1]
```diff
@@ -23,3 +24,16 @@ def test_content_with_unknow_constants(self):
...
+            Content.objects.create(name="content", content="content")
+            Content.objects.create(name="content", content="content")
+
+
+class ConstantTestCase(TestCase):
+
+    def test_create_two_constants_with_same_name(self) -> None:
+        with self.assertRaises(IntegrityError):
+            Constant.objects.create(name="const", value="1")
+            Constant.objects.create(name="const", value="1")
```
Comment (Lymkwi) [inline insalan/cms/tests.py:39]: maybe just for the sake of not missing something obvious these two (and the two above) should also have different *values*

**Review by Lymkwi — CHANGES_REQUESTED** [review]

**Review by Lugrim — CHANGES_REQUESTED** [review]
For consistency, please add docstrings and switch to double quotes (you should be able to do that by running `black`)

File: insalan/cms/tests.py (hunk 1 above)
Comment (ShiroUsagi-san) [reply insalan/cms/tests.py]: fixed in #0dd7c

File: insalan/cms/models.py [hunk 2]
```diff
@@ -8,25 +8,37 @@ def constant_definition_validator(content: str):
...
-    excess_constants = constant_list - p_constants.intersection(constant_list) # Get the constants in the content but not defined
+    excess_constants = constant_list - p_constants.intersection(
+        constant_list
+    )  # Get the constants in the content but not defined
 
     if len(excess_constants) > 0:
         raise ValidationError(
-                _(f"Des constantes non définies sont utilisées: {', '.join(sorted(excess_constants))}"))
+            _(
+                f"Des constantes non définies sont utilisées: {', '.join(sorted(excess_constants))}"
```
Comment (Lymkwi) [inline insalan/cms/models.py:26]: This kind of messes up the translation string if I remember correctly, but I'm just commenting on it here to make a mental note to go check later

File: insalan/cms/tests.py [hunk 3]
```diff
@@ -3,27 +3,34 @@
...
-        """ test that we can create a content using constant """
+        """test that we can create a content using constant"""
         Constant.objects.create(name="a", value="je suis la valeur a")
         Constant.objects.create(name="b", value="je suis la valeur b")
-        Content.objects.create(name="content_a", content="j'appelle la constante ${a} et la constante ${b}")
+        Content.objects.create(
+            name="content_a", content="j'appelle la constante ${a} et la constante ${b}"
+        )
 
     def test_content_with_unknow_constants(self):
```
Comment (Lymkwi) [inline insalan/cms/tests.py:18]: I think the following three methods are those @Lugrim wanted you to document a tiny bit, even just for the sake of consistency.

**Review by Lymkwi — CHANGES_REQUESTED** [review]

File: insalan/cms/tests.py (hunk 3 above)
Comment (ShiroUsagi-san) [reply insalan/cms/tests.py:18]: fixed by [092dd48](https://github.com/InsaLan/backend-insalan.fr/pull/92/commits/092dd488d5ec8dfa7a524c08747fc6d20c6e2dc2)

File: insalan/cms/models.py (hunk 2 above)
Comment (ShiroUsagi-san) [reply insalan/cms/models.py:26]: what's the best way to deal with this kind of strings ?

Comment (ShiroUsagi-san) [reply insalan/cms/models.py]: changed to format() in [240d910](https://github.com/InsaLan/backend-insalan.fr/pull/92/commits/240d910b0ca06445987028748b3ac0c3586174e2)

File: insalan/cms/tests.py [hunk 4]
```diff
@@ -3,37 +3,48 @@
...
-        Content.objects.create(name="content", content="je suis ${a} mais aussi une variable ${random}")
+        Content.objects.create(
+            name="content", content="je suis ${a} mais aussi une variable ${random}"
+        )
         self.assertRaises(ValidationError)
 
     def test_display_undefined_constant_list(self):
-        Content.objects.create(name="content", content="je test ${inconnue} mais aussi une variable ${random}")
-        self.assertRaisesMessage(ValidationError, _("Des constantes non définies sont utilisées: inconnue, random"))
+        """ Test that a ValidationError raised by the usage of unknown constants give the correct list of undefined constants
```
Comment (Lymkwi) [inline insalan/cms/tests.py:27]: you can add a line feed after the `"""` and put the docstring on its own line

File: insalan/cms/tests.py [hunk 5]
```diff
@@ -3,37 +3,48 @@
...
-            Content.objects.create(name="content", content="content")
-            Content.objects.create(name="content", content="content")
+            Content.objects.create(name="content", content="content1")
+            Content.objects.create(name="content", content="content2")
 
 
 class ConstantTestCase(TestCase):
-
     def test_create_two_constants_with_same_name(self) -> None:
+        """ Test that the name unicity of a constant is checke """
```
Comment (Lymkwi) [inline insalan/cms/tests.py:47]: typo

**Review by Lymkwi — CHANGES_REQUESTED** [review]
Almost! oh and no need to add spaces between the docstring and the triple quotes

File: insalan/cms/tests.py (hunk 4 above)
Comment (ShiroUsagi-san) [reply insalan/cms/tests.py:27]: I have to confess: I let black auto-fixed the code :roll_eyes:

**Review by Lymkwi — APPROVED** [review]
Goddamn, you've done it again Mahal! Congrats on the good work

**Review by Lugrim — APPROVED** [review]
Great work!

**Pull request merged**